cli/mender-artifact/partition.go

```diff
@@ -26,25 +27,106 @@ import (
 	"github.com/pkg/errors"
 )
 
-// PartitionReadWriteClosePacker wraps io.ReadWriteCloser with a Repack method
-type PartitionReadWriteClosePacker interface {
-	io.ReadWriteCloser
-	Repack() error
-}
-
 type partition struct {
 	offset string
 	size   string
 	path   string
 	name   string
 }
 
-// partitions is a wrapper around partitionFile, so that
-// a write is duplicated to both partitions' files during a write
-type partitions []partitionFile
+// sdimgFile is a wrapper around partitionFile, so that
+// a write is duplicated to both sdimgFile' files during a write
+type sdimgFile []io.ReadWriteCloser
+
+func stripSDimgBootExtensions(fpath string) string {
+	// Map /boot/{efi,grub} to /boot
+	if strings.HasPrefix(fpath, "/boot/efi/") {
+		return strings.TrimPrefix(fpath, "/boot/efi/")
+	} else if strings.HasPrefix(fpath, "/boot/grub") {
+		return strings.TrimPrefix(fpath, "/boot/grub/")
+	}
+	// /boot prefix
+	return strings.TrimPrefix(fpath, "/boot/")
```

Comment:
Shouldn't this be `/uboot`? `/boot` is not the boot partition, it is the `boot` directory on the root partition. The three possibilities are:

* `/uboot`
* `/boot/efi`
* `/boot/grub`

There are a lot of places in the code this might need changing.